packages/core/src/actions/smartAccount/buildUserOperation.ts

```diff
@@ -34,11 +34,14 @@ export const buildUserOperation: <
     );
   }
 
+  const hexaNonce =
+    "0x" + (await account.getNonce(overrides?.nonceKey)).toString(16);
+
   return _runMiddlewareStack(client, {
     uo: {
       initCode: account.getInitCode(),
       sender: account.address,
-      nonce: account.getNonce(overrides?.nonceKey),
+      nonce: hexaNonce,
```

Comment:
is this necessary @nsnull0?  why was this failing before?

Reply:
thank you for follow up.
to be honest i am not sure where is the best place to fix it, i just come up with this idea when looking at the aa-sdk package. 

i was testing `buildUserOperation` method.
and then it gives me error nonce should be hexadecimal.
I was able to make run through it by handwrite manually
```
batchUoStruct.nonce = "0x" + batchUoStruct.nonce.toString(16)
```
then it is succesfully sent userOP.

ref:
![Screenshot 2024-04-09 at 13 09 07](https://github.com/alchemyplatform/aa-sdk/assets/26643922/abe0eab9-2e59-4ad2-a107-201992140008)

Reply:
you can also just do `toHex(account.getNonce())`. `toHex` is exported from `viem`

Reply:
i'll have a look, thank you for info..